deadlines/src/main/java/com/palantir/deadlines/Deadlines.java

```diff
@@ -114,39 +119,107 @@ public static <T> void encodeToRequest(
         if (stateDeadline == null) {
             // use proposedDeadline
-            checkExpiration(proposedDeadlineNanos, false, false, false);
+            checkExpiration(
+                    proposedDeadlineNanos,
+                    false,
+                    false,
+                    false,
+                    // don't bother with enforcement here
+                    // if we don't have any existing deadline state, then we're merely checking that the caller
+                    // didn't pass in a negative deadline to this method
+                    // eventually, enforcement will be on by default anyway
+                    false);
             adapter.setHeader(
                     request, DeadlinesHttpHeaders.EXPECT_WITHIN, durationToHeaderValue(proposedDeadlineNanos));
         } else {
             // use the minimum of proposedDeadline and the one read from state
             long remainingStateDeadlineNanos = stateDeadline.remainingNanos(getClockNanoTime());
+            Enforcement enforcement = stateDeadline.enforcement();
+            boolean enforced = enforcement == Enforcement.ENFORCE;
             if (proposedDeadlineNanos <= remainingStateDeadlineNanos) {
                 boolean proposedDeadlineAlreadyExpired = proposedDeadline.isNegative() || proposedDeadline.isZero();
                 checkExpiration(
                         proposedDeadlineNanos,
                         false,
                         stateDeadline.disablePropagation(),
-                        proposedDeadlineAlreadyExpired);
+                        proposedDeadlineAlreadyExpired,
+                        enforced);
                 if (!stateDeadline.disablePropagation()) {
                     adapter.setHeader(
                             request, DeadlinesHttpHeaders.EXPECT_WITHIN, durationToHeaderValue(proposedDeadlineNanos));
+                    encodeEnforcement(request, adapter, enforcement);
                 }
             } else {
                 checkExpiration(
                         remainingStateDeadlineNanos,
                         stateDeadline.internal(),
                         stateDeadline.disablePropagation(),
-                        stateDeadline.alreadyExpired());
+                        stateDeadline.alreadyExpired(),
+                        enforced);
                 if (!stateDeadline.disablePropagation()) {
                     adapter.setHeader(
                             request,
                             DeadlinesHttpHeaders.EXPECT_WITHIN,
                             durationToHeaderValue(remainingStateDeadlineNanos));
+                    encodeEnforcement(request, adapter, enforcement);
                 }
             }
         }
     }
 
+    private static <T> void encodeEnforcement(
+            T request, RequestEncodingAdapter<? super T> adapter, Enforcement enforcement) {
+        if (enforcement == Enforcement.ENFORCE) {
```

Comment:
Thoughts on using a switch-expression here so the compiler will tell us when we don't have exhaustive coverage?